src/client/datascience/jupyter/kernels/kernelDependencyService.ts

```diff
@@ -92,24 +120,31 @@ export class KernelDependencyService implements IKernelDependencyService {
             interpreter.displayName || interpreter.path,
             ProductNames.get(Product.ipykernel)!
         );
-        const installerToken = wrapCancellationTokens(token);
-        // If there's no UI, then cancel installation.
-        if (disableUI) {
-            return KernelInterpreterDependencyResponse.cancel;
-        }
         sendTelemetryEvent(Telemetry.PythonModuleInstal, undefined, {
             action: 'displayed',
             moduleName: ProductNames.get(Product.ipykernel)!
         });
+        const promptCancellationPromise = createPromiseFromCancellation({
+            cancelAction: 'resolve',
+            defaultValue: undefined,
+            token
+        });
         const installPrompt = isModulePresent ? Common.reInstall() : Common.install();
+        const selectKernel = DataScience.selectKernel();
         const selection = this.isCodeSpace
             ? installPrompt
-            : await Promise.race([this.appShell.showErrorMessage(message, installPrompt), promptCancellationPromise]);
+            : await Promise.race([
+                  this.appShell.showErrorMessage(message, installPrompt, selectKernel),
+                  promptCancellationPromise
+              ]);
         if (installerToken.isCancellationRequested) {
             return KernelInterpreterDependencyResponse.cancel;
         }
 
-        if (selection === installPrompt) {
+        if (selection === selectKernel) {
+            this.commandManager.executeCommand('notebook.selectKernel').then(noop, noop);
```

Comment:
Should this have the same logic based on webview/native?

Reply:
Actually this seems like it will call execute command twice? Doesn't handleDependencyResponse also do this?

Reply:
Fixed when writing the tests (it wasn't supposed to be there)